src/nnet3/nnet-simple-component.cc

```diff
@@ -2140,6 +2141,7 @@ void ConstantFunctionComponent::Backprop(
 }
 
 void ConstantFunctionComponent::Read(std::istream &is, bool binary) {
+  ReadUpdatableCommon(is, binary);  // Read the opening tag and learning rate
   ExpectOneOrTwoTokens(is, binary, "<ConstantFunctionComponent>",
```

Comment:
This should just be ExpectToken, because by this point you've already read the opening tag.